src/storage/secondary/rowset/rowset_iterator.rs

```diff
@@ -265,14 +275,46 @@ impl RowSetIterator {
                     arrays[id] = Some(array);
                 }
             }
+            // For now, we only support range-filter scan by first column.
+            if !self.end_keys.is_empty() && id == 0 {
+                let end_key = &self.end_keys[0];
+                let array = arrays[0].as_ref().unwrap();
+                let len = array.len();
+
+                // Todo: only suppor range-filter scan by sort key type of int32, support other type
+                // later.
+                if end_key - &array.get(len - 1) < DataValue::Int32(0) {
```

Comment:
Directly compare by `end_key < &array.get(len - 1)`?

Reply:
Because I found that we implemented  addition, subtraction, multiplication and division between `DataValue`（like int32， int64). so i try reuse the logic to avoid many   forced conversions

Reply:
This thing is a little weird. I'll think of something else.😂